lib/gtf/genome.cc

```diff
@@ -76,48 +76,48 @@ int genome::read_gz(const string &file)
 	return 0;
 }
 
-// int genome::read(const string &file)
-// {
-// 	if(file == "") return 0;
+int genome::read(const string &file)
+{
+	if(file == "") return 0;
 
-// 	ifstream fin(file.c_str());
-// 	if(fin.fail())
-// 	{
-// 		printf("open file %s error\n", file.c_str());
-// 		return 0;
-// 	}
+	ifstream fin(file.c_str());
+	if(fin.fail())
+	{
+		printf("open file %s error\n", file.c_str());
+		return 0;
+	}
 
-// 	char line[102400];
+	char line[102400];
 	
-// 	genes.clear();
-// 	g2i.clear();
-// 	while(fin.getline(line, 102400, '\n'))
-// 	{
-// 		item ge(line);
-// 		if(g2i.find(ge.gene_id) == g2i.end())
-// 		{
-// 			gene gg;
-// 			if(ge.feature == "transcript") gg.add_transcript(ge);
-// 			else if(ge.feature == "exon") gg.add_exon(ge);
-// 			g2i.insert(pair<string, int>(ge.gene_id, genes.size()));
-// 			genes.push_back(gg);
-// 		}
-// 		else
-// 		{
-// 			int k = g2i[ge.gene_id];
-// 			if(ge.feature == "transcript") genes[k].add_transcript(ge);
-// 			else if(ge.feature == "exon") genes[k].add_exon(ge);
-// 		}
-// 	}
+	genes.clear();
+	g2i.clear();
+	while(fin.getline(line, 102400, '\n'))
```

Comment:
Replace 102400 with `sizeof(line)`. Dangerous to repeat a hard-coded constant. (I know it is not your code originally).

Reply:
The same pattern may exists in another function.